src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -289,8 +324,7 @@ public void postInject() {
 
         logger.dfltKv("ThingName", (Supplier<String>) () ->
                 Coerce.toString(deviceConfiguration.getThingName()));
-
-        executorService.submit(() -> {
+        executorService.execute(() -> {
```

Comment:
save the future, make it a singleton. What if multiple changes come in? This is going to end up executing several times.